Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -15256,6 +15256,31 @@ void SpeculativeJIT::compileMapSet(Node* node)
 
 void SpeculativeJIT::compileWeakMapGet(Node* node)
 {
+    if (node->child2().useKind() == CellUse) {
+        SpeculateCellOperand weakMap(this, node->child1());
+        SpeculateCellOperand key(this, node->child2());
+        SpeculateInt32Operand hash(this, node->child3());
+
+        GPRReg weakMapGPR = weakMap.gpr();
+        if (node->child1().useKind() == WeakMapObjectUse)
+            speculateWeakMapObject(node->child1(), weakMapGPR);
+        else
+            speculateWeakSetObject(node->child1(), weakMapGPR);
+        GPRReg keyGPR = key.gpr();
+        GPRReg hashGPR = hash.gpr();
+
+        flushRegisters();
+        JSValueRegsFlushedCallResult result(this);
+        JSValueRegs resultRegs = result.regs();
+        if (node->child1().useKind() == WeakMapObjectUse)
+            callOperation(operationWeakMapGet, resultRegs, TrustedImmPtr(&vm()), weakMapGPR, keyGPR, hashGPR);
+        else
+            callOperation(operationWeakSetGet, resultRegs, TrustedImmPtr(&vm()), weakMapGPR, keyGPR, hashGPR);
+        m_jit.exceptionCheck();
+        jsValueResult(resultRegs, node);
+        return;
+    }
```

Comment:
ditto as comment about FTL code, why can't we run the fast path here?
If it's some bad cell value, like a string or a register symbol, we won't find it in the hash map 
Maybe just a comment to explain why it's ok

Reply:
Yeah, right. Removed this.